pkgs/os-specific/linux/cryptsetup/default.nix

```diff
@@ -13,8 +21,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-/A35RRiBciZOxb8dC9oIJk+tyKP4VtR+upHzH+NUtQc=";
   };
 
-  # Disable 4 test cases that fail in a sandbox
-  patches = [ ./disable-failing-tests.patch ];
+  patches = [
+    # Disable 4 test cases that fail in sandbox
+    ./disable-failing-tests.patch
+    # Allow reading tokens from a relative path, see #167994
+    ./relative-token-path.patch
```

Comment:
I think upstream was intentionally using absolute paths here to avoid some security issue? @dasJ ?
Maybe it would be good to recall what they were and see if he it any case where this would be relevant.

Reply:
I'd like to agree with @Mic92 that relative path may open a bad door when you may inject wrong library.

Reply:
I'm waiting for feedback from @dasJ...

Reply:
Can you describe the attack vector?

Reply:
And about that commit: How do you break the cyclic dependency between systemd and cryptsetup in that case?

Reply:
@dasJ all vectors that I see right now requires an equiualent of root access. If the bad guy had it => he may had much worst things than simple inject some `.so`;

Regarding commit: my point that original idea commit wasn't related to security at all => it should be safe to apply current patch.

And I agree that the patch is the way to break the cyclic depenency.

Reply:
"Attack vectors" people told me:

* The tool running as root (for example via sudo) could load an unprivileged file which is not relevant because it's not our job to check the environment, the user could set LD_PRELOAD as well. Also sudo should drop these
* systemd-cryptsetup could be used as a setuid binary (why would anyone do that in the first place), it might load .so files from LD_LIBRARY_PATH. Irrelevant as using setuid ignores `LD_LIBRARY_PATH` via the secure-execution mode

Reply:
@dasJ please make `relative-token-path.patch` `git format-patch`-style, and add the explanation in the "commit message". That way it won't need to be explained over and over again, but lives alongside the patch.

Reply:
Why me? This is neither my PR nor my commit

Reply:
Should be safe than.

Reply:
We should have some longer explanation on why we undo this "security feature" on NixOS, to avoid having to discuss about this again and again. Having it in the commit message of the patch would be a viable solution. @Kranzes, can you add it?